lib/compliance/blackduck.js

```diff
@@ -0,0 +1,244 @@
+            continue
+          }
+          let license = this._getLicense(packageLicenseDatum.licenseSources)
+          versionData.push({version: packageLicenseDatum.version, name: packageLicenseDatum.name, license})
+        }
+        versionData = _.sortBy(versionData, ['name', 'version'])
+
+        return this.writeFile(npmOutputFilePath, JSON.stringify({'schema-version': 1, data: versionData}, null, '  '))
+          .then(() => {
+            return `successfully wrote ${npmOutputFilePath}`
+          })
+          .catch((error) => {
+            logger.log(`(1) ERROR: writing ${npmOutputFilePath}`, error)
+            throw error
+          })
+      })
+  },
+
+  /**
+   * Get a list of repositories that don't contain code we care about
+   * @param {String} cwd - string for current-working-directory
+   * @param {String} ourRepoName - the name of our own repos
+   * @returns {Promise} A promise for a string of repository URLs
+   */
+  getPackageData (cwd, ourRepoName) {
+    console.log('...getting package data')
+    const publicRegex = /\s+"(ember\-frost\-\S+)"/
+    const privateRegex = /\s+"(frost\-\S+)"/
+    const filename = path.join(cwd, 'package.json')
+    return this.readFile(filename, 'utf8')
+      .then((result) => {
+        const data = _.filter(result.split('\n').map((line) => {
+          let name
+          let matches = publicRegex.exec(line)
+          if (matches !== null) {
+            name = matches[1]
+            return `https://github.com/ciena-frost/${name}.git`
+          }
+          matches = privateRegex.exec(line)
+          if (matches !== null) {
+            name = matches[1]
+            if (name === ourRepoName) {
+              return undefined
+            }
+            return `https://bitbucket.ciena.com/scm/bp_frost/${name}.git`
+          }
+        }))
+        return data.sort().join('\n') + '\n'
+      })
+  },
+
+  /**
+   * Make sure the path exists and that it's a directory
+   * @param {String} directoryPath - a directory for where we're putting stuff
+   * @returns {Promise} a promise for the directory existing
+   */
+  ensureDirectory (directoryPath) {
+    console.log('...checking directory')
+    return this.stat(directoryPath)
+      .then((stats) => {
+        if (stats.isDirectory()) {
+          return true
+        }
+        throw new Error(`${directoryPath} exists but is not a directory!`)
+      })
+      .catch(() => {
+        return this.mkdir(directoryPath)
+      })
+  },
+
+  /**
+   * Run processing for blackduck and output to the directory specified
+   * @param {String} cwd - string for current-working-directory
+   * @param {String} directoryPath - the place to dump our files
+   * @param {String} ourRepoName - the name of our own repos
+   * @returns {Promise} - a promise for everything to be done
+   */
+  main (cwd, directoryPath, ourRepoName) {
+    const npmOutputFilePath = path.join(directoryPath, 'js_requirements.json')
```

Comment:
The filename to generate should be another option in `.pr-bumper.json` (can default to `js_requirements.json` if we want).